services/paymaster/src/db/config.rs

```diff
@@ -70,74 +114,26 @@ pub async fn load_config() -> Result<Config, sqlx::Error> {
         {
-            if version == "v0" {
-                // Extract the whitelisted_programs array as Vec<String>
-                let strings: Vec<String> = match transaction_variation.0 {
-                    JsonValue::Array(arr) => arr
-                        .iter()
-                        .filter_map(|v| v.as_str().map(|s| s.to_string()))
-                        .collect(),
-                    _ => {
-                        tracing::warn!(
-                            ?domain,
-                            ?version,
-                            ?name,
-                            "Missing or invalid whitelisted_programs field",
-                        );
-                        continue;
-                    }
-                };
-
-                // Convert Vec<String> to Vec<Pubkey>
-                let whitelisted_programs: Vec<Pubkey> = match strings
-                    .into_iter()
-                    .map(|s| Pubkey::from_str(&s))
-                    .collect::<Result<Vec<_>, _>>()
-                {
-                    Ok(pubkeys) => pubkeys,
-                    Err(err) => {
-                        tracing::warn!(
-                            ?err,
-                            ?domain,
-                            ?version,
-                            ?name,
-                            "Failed to parse pubkeys from whitelisted_programs",
-                        );
-                        continue;
-                    }
-                };
-
-                transaction_variation_fin = TransactionVariation::V0(VariationProgramWhitelist {
+            transaction_variation_fin = match version.as_str() {
+                "v0" => handle_transaction_variation_v0(transaction_variation, name)?,
+                "v1" => handle_transaction_variation_v1(
+                    transaction_variation,
                     name,
-                    whitelisted_programs,
-                });
-            } else {
-                let instructions = match serde_json::from_value(transaction_variation.0) {
-                    Ok(tv) => tv,
-                    Err(err) => {
-                        // log and skip this row instead of failing the whole query
-                        tracing::warn!(
-                            ?err,
-                            ?domain,
-                            ?version,
-                            ?name,
-                            "Skipping row with invalid transaction_variation",
-                        );
-                        continue;
-                    }
-                };
-
-                transaction_variation_fin =
-                    TransactionVariation::V1(VariationOrderedInstructionConstraints {
-                        name,
-                        instructions,
-                        max_gas_spend: max_gas_spend.map(|v| v as u64).unwrap_or(0),
-                    });
-            }
-
+                    max_gas_spend as u64,
```

Comment:
casting via `as` is not the best practice because the conversion isn't checked. e.g. for the `i64 -> u64` conversion here, the `as` keyword will wrap negative `i64`s into large positive `u64`s. instead i would try `u64::try_from(max_gas_spend)?` with appropriate error mapping and propagation